modules/core/src/matrix.cpp

```diff
@@ -7,6 +7,375 @@
 
 namespace cv {
 
+std::string layoutToString(DataLayout layout)
+{
+    return
+        layout == DATA_LAYOUT_ND ? "ND" :
+        layout == DATA_LAYOUT_NCHW ? "NCHW" :
+        layout == DATA_LAYOUT_NHWC ? "NHWC" :
+        layout == DATA_LAYOUT_BLOCK ? "NC1HWC0" :
+        layout == DATA_LAYOUT_NCDHW ? "NCDHW" :
+        layout == DATA_LAYOUT_NDHWC ? "NDHWC" :
+        layout == DATA_LAYOUT_PLANAR ? "PLANAR" :
+        layout == DATA_LAYOUT_UNKNOWN ? "Unknown" : "???";
```

Comment:
Assert? Invalid layout value means user error or memory corruption. Both of them should stop the program.